src/azul/tdr.py

```diff
@@ -38,168 +21,77 @@
     RequirementError,
     cached_property,
     config,
-    require,
-)
-from azul.bigquery import (
-    AbstractBigQueryAdapter,
-    BigQueryAdapter,
-    BigQueryRow,
-    BigQueryRows,
 )
 from azul.dss import (
     shared_credentials,
 )
-from azul.indexer import (
-    Bundle,
-    BundleFQID,
-)
 from azul.types import (
     JSON,
 )
-from azul.uuids import (
-    validate_uuid_prefix,
-)
 
 log = logging.getLogger(__name__)
 
 
 @attr.s(frozen=True, auto_attribs=True, kw_only=True)
-class BigQueryDataset:
+class TDRSource:
     project: str
     name: str
     is_snapshot: bool
 
+    _type_dataset = 'dataset'
+
+    _type_snapshot = 'snapshot'
+
     @classmethod
-    def parse(cls, tdr_target: str) -> 'BigQueryDataset':
+    def parse(cls, source: str) -> 'TDRSource':
         """
         Construct an instance from its string representation, using the syntax
-        'tdr:{project}:{target_type}/{target_name}'.
-
-        >>> BigQueryDataset.parse('tdr:my_project:snapshot/snapshot1')
-        BigQueryDataset(project='my_project', name='snapshot1', is_snapshot=True)
-
-        >>> BigQueryDataset.parse('tdr:my_project:dataset/dataset1')
-        BigQueryDataset(project='my_project', name='datarepo_dataset1', is_snapshot=False)
-
-        >>> BigQueryDataset.parse('foo:my_project:dataset/dataset1')
+        'tdr:{project}:{source_type}/{source_name}'.
+
+        >>> s = TDRSource.parse('tdr:foo:snapshot/bar')
+        >>> s
+        TDRSource(project='foo', name='bar', is_snapshot=True)
```

Comment:
Note the gibberish names like `foo` which cleanly distinguishes them as play data and exaggerates the distinction to names used in the code under test.